sandbox/src/mock-api.js

```diff
@@ -28,133 +28,176 @@ app.get("/observation", (req, res) => {
   }
 
-  let responseBody = {
-    resourceType: "Bundle",
-    id: "1a420b58-5b91-4706-9a7c-90028ca79ff3",
-    meta: {
-      lastUpdated: "2021-01-20T12:46:07.830+00:00",
-    },
-    type: "searchset",
-    total: 2,
-    entry: [
-      {
-        fullUrl:
-          "https://test-results-api.nhs.uk/R4/Observation/eb89a392-5b20-11eb-ae93-0242ac130002",
-        resource: {
-          resourceType: "Observation",
-          id: "eb89a392-5b20-11eb-ae93-0242ac130002",
-          identifier: [
-            {
-              system: "urn:oid:",
-              value: "eb89a392-5b20-11eb-ae93-0242ac130002",
-            },
-          ],
-          status: "final",
-          subject: {
-            reference: "Patient/123456",
+  const nhsNumberNotFoundPattern = "^https:\/\/fhir.nhs.uk\/Id\/nhs-number[|]1000000000$";
+  let nhsNumberNotFound = new RegExp(nhsNumberNotFoundPattern);
+
+  if (req.query["patient.identifier"] & nhsNumberNotFound.test(req.query["patient.identifier"])) {
+    let responseBody = {
+      resourceType: "Bundle",
+      id: "1a420b58-5b91-4706-9a7c-90028ca79ff3",
+      type: "searchset",
+      total: 0,
+      entry: [],
+    };
+  
+    res.send(responseBody);
+  } else {
+    let responseBody = {
+      resourceType: "Bundle",
+      id: "1a420b58-5b91-4706-9a7c-90028ca79ff3",
+      meta: {
+        lastUpdated: "2021-01-20T12:46:07.830+00:00",
+      },
+      type: "searchset",
+      total: 2,
+      entry: [
+        {
+          fullUrl:
+            "https://test-results-api.nhs.uk/R4/Observation/eb89a392-5b20-11eb-ae93-0242ac130002", //ToDo: Check if it was removed from the real API
+          resource: {
+            resourceType: "Observation",
+            id: "ASD32145123",
             identifier: [
               {
-                system: "https://fhir.nhs.uk/Id/nhs-number",
-                value: req.query["patient.identifier"].split("|")[1],
+                system: "https://fhir.nhs.uk/Id/SpecimenId",
+                value: "ASD32145123",
               },
             ],
-          },
-          effectiveDateTime: "2020-09-23T13:00:08.476+00:00",
-          valueCodeableConcept: {
-            coding: [
-              {
-                system: "http://snomed.info/sct",
-                code: "1240581000000104",
-                display:
-                  "SARS-CoV-2 (severe acute respiratory syndrome coronavirus 2) detection result positive",
-              },
-            ],
-            text: "SARS-CoV-2-ORGY",
-          },
-          method: {
-            coding: [
-              {
+            status: "final",
+            subject: {
+              reference: "Patient/3fa85f64-5717-4562-b3fc-2c963f66afa6", //ToDo: Change value to Patient/DPSsubjectId
+              type: "Patient", //ToDo: Missing in the real API
+              identifier: [
+                {
+                  system: "https://fhir.nhs.uk/Id/nhs-number",
+                  value: req.query.patient,
```

Comment:
shouldn't we be using `req.query["patient.identifier"].split("|")[1]` here?

Reply:
Above it's been changed to SpecimenID - Marcin, could you do the same modification here?